src/encoded/types/library.py

```diff
@@ -20,3 +37,86 @@ class Library(SubmittedItem):
     item_type = "library"
     schema = load_schema("encoded:schemas/library.json")
     embedded_list = _build_library_embedded_list()
+
+    class Collection(Item.Collection):
+        pass
+
+def validate_molecule_specific_assay(context,request):
+    """Check that analyte.molecule includes the correct molecule for molecule-specific assays.
+    
+    The assays with `valid_molecules` property may need to be updated as new techologies come out 
+    or are added to the portal.
+    """
+
+    data = request.json
+    if 'analytes' in data:
+        molecules = []
+        for analyte in data['analytes']:
+            molecules+=get_item_or_none(request, analyte, 'analytes').get("molecule",[])
+        assay = get_item_or_none(request,data['assay'],'assays')
+        valid_molecules = assay.get("valid_molecules",[])
+        if valid_molecules:
+            overlap = list(set(molecules) & set(valid_molecules))
+            if not overlap:
+                msg = f"Assay {assay} is specific to molecules: {valid_molecules}."
+                return request.errors.add('body', 'Library: invalid links', msg)
+        return request.validated.update({})
+    
+
+def validate_assay_specific_properties(context,request):
+    """Check that assay is appropriate for assay-specific properties.
+    
+    Assay-specific properties are in `ASSAY_DEPENDENT`
+    """
+    data = request.json
+    if 'assay' in data:
+        for property in ASSAY_DEPENDENT.keys():
+            if data.get(property,""):
```

Comment:
Instead of doing a get for property value I think with these properties that don't have a default you could altenatively use:
`if property in data:` to check for its presence as you aren't using the value here?